app/controllers/CheckDetailsController.scala

```diff
@@ -73,50 +74,48 @@ class CheckDetailsController @Inject()(
                                 )(implicit hc: HeaderCarrier, request: DataRequest[AnyContent]): Future[Result] = {
 
     logger.info(s"Valid origin: $origin")
-
-    val pdvRequest = PDVRequest(
-      request.credId.getOrElse(EmptyString),
-      request.session.data.getOrElse("sessionId", EmptyString)
-    )
-
-    val result: Try[Future[Result]] = Try {
-      val processData = for {
-        updatedAnswers <- Future.fromTry(request.userAnswers.set(OriginCacheable, origin.getOrElse("None")))
-        _ <- sessionRepository.set(updatedAnswers)
-        pdvData <- personalDetailsValidationService.getPDVData(pdvRequest)
-        idData  <- individualDetailsService.getIdData(pdvData)
-        sessionWithNINO = request.session + ("nino" -> getNinoFromPDVResponse(pdvData))
-      } yield (pdvData, idData, sessionWithNINO) match {
-        case (pdvData, Left(idData), sessionWithNINO)  => checkDetailsFailureJourney(pdvData, idData, mode, sessionWithNINO, origin)
-        case (pdvData, Right(idData), sessionWithNINO) => checkDetailsSuccessJourney(pdvData, idData, mode, sessionWithNINO, origin)
-        case _                                         => checkDetailsMatchingFailedWithUnknownIssue(mode)
-      }
-      processData.recover {
-        case ex: Exception =>
-          logger.error(s"An error occurred in process data: ${ex.getMessage}")
-          Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode))
-      }
-    }
-    result match {
-      case Success(res) => res
-      case Failure(ex) =>
-        logger.error(s"An error occurred, redirecting: ${ex.getMessage}")
-        Future(Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)))
+    val pdvRequest = PDVRequest(request.credId.getOrElse(EmptyString), request.session.data.getOrElse("sessionId", EmptyString))
+
+    val processData:Future[Result] = for {
+      updatedAnswers <- Future.fromTry(request.userAnswers.set(OriginCacheable, origin.getOrElse("None")))
+      _ <- sessionRepository.set(updatedAnswers)
+      pdvData <- personalDetailsValidationService.getPDVData(pdvRequest)
```

Comment:
I also think we should not tie PDV and 1694 calls together. Again I have an alternate approach that I want to share with the team in a mob session.

Reply:
yes I too agree. If PDV fails, no need to call API1694 with empty data. We can avoid it. We had this plan at our first E2E testing but we are worried to refactor it at this stage.